core/azure-core/src/main/java/com/azure/core/amqp/exception/AmqpException.java

```diff
@@ -0,0 +1,98 @@
+ */
+public class AmqpException extends AzureException {
+    private static final long serialVersionUID = -3654294093967132325L;
+
+    private ErrorContext errorContext;
+    private final boolean isTransient;
+    private final ErrorCondition errorCondition;
+
+    /**
+     * Initializes a new instance of the AmqpException class.
+     *
+     * @param isTransient A boolean indicating if the exception is a transient error or not. If true, then the request
+     * can be retried; otherwise not.
+     * @param description Text containing any supplementary details of the exception.
+     */
+    public AmqpException(boolean isTransient, String description) {
+        this(isTransient, null, description);
+    }
+
+    /**
+     * Initializes a new instance of the AmqpException class.
+     *
+     * @param isTransient A boolean indicating if the exception is a transient error or not. If true, then the request
+     * can be retried; otherwise not.
+     * @param errorCondition The symbolic value indicating the error condition.
+     * @param description Text containing any supplementary details not indicated by the condition field. This text can
+     * be logged as an aid to resolving issues.
+     */
+    public AmqpException(boolean isTransient, ErrorCondition errorCondition, String description) {
+        super(description);
+        this.errorCondition = errorCondition;
+        this.isTransient = isTransient;
+    }
+
+    @Override
+    public String getMessage() {
+        final String baseMessage = super.getMessage();
+
+        if (this.errorContext == null) {
+            return super.getMessage();
+        }
+
+        return !ImplUtils.isNullOrEmpty(baseMessage)
+            ? String.format(Locale.US, "%s, %s[%s]", baseMessage, "errorContext", errorContext.toString())
+            : String.format(Locale.US, "%s[%s]", "errorContext", errorContext.toString());
+    }
+
+    /**
+     * A boolean indicating if the exception is a transient error or not.
+     *
+     * @return returns true when user can retry the operation that generated the exception without additional intervention.
+     */
+    public boolean getIsTransient() {
+        return this.isTransient;
+    }
+
+    /**
+     * Gets the ErrorCondition for this exception.
+     *
+     * @return The ErrorCondition for this exception, or {@code null} if nothing was set.
+     */
+    public ErrorCondition getErrorCondition() {
+        return this.errorCondition;
+    }
+
+    /**
+     * Gets the context for this exception.
+     *
+     * @return The context for this exception.
+     */
+    public ErrorContext getContext() {
+        return this.errorContext;
+    }
+
+    /**
+     * Sets the context for this exception.
+     *
+     * @param errorContext The context for this exception.
+     */
+    public void setContext(ErrorContext errorContext) {
+        this.errorContext = errorContext;
+    }
```

Comment:
This is weird, current implementation bubbles it up to the Sender code which sets it because it has the SenderErrorContext. I wanted to remove this mutability but I have yet to handle the error handling.